src/test/java/org/lisoft/lsml/model/item/WeaponTest.java

```diff
@@ -96,13 +97,13 @@ public void testConstruction() {
         assertEquals(cooldown, cut.getCoolDown(null), 0.0);
         assertSame(aRangeProfile, cut.getRangeProfile());
 
-        assertEquals(aRoundsPerShot, cut.getAmmoPerPerShot());
+        assertEquals(aRoundsPerShot, cut.getRoundsPerShot());
         assertEquals(aDamagePerProjectile * aProjectilesPerRound * aRoundsPerShot, cut.getDamagePerShot(), 0.0);
         assertEquals(projectileSpeed, cut.getProjectileSpeed(null), 0.0);
         assertEquals(aGhostHeatGroupId, cut.getGhostHeatGroup());
         assertEquals(aGhostHeatMultiplier, cut.getGhostHeatMultiplier(), 0.0);
         assertEquals(ghostHeatMaxFreeAlpha, cut.getGhostHeatMaxFreeAlpha(null));
-        assertEquals(cooldown + aVolleyDelay * (aRoundsPerShot - 1), cut.getExpectedFiringPeriod(null), 0.0);
```

Comment:
A weapon without type only has cooldown.  So, I agree, the test does not really apply at the weapon level, because there is no weapon (other than AMS) that is not either AmmoWeapon or EnergyWeapon.